### ISSUE TITLE
Add some scaladoc about ContextPropagation?

### ISSUE THREAD

**Issue by He-Pin** [akka/akka#31621]
![image](https://user-images.githubusercontent.com/501740/192240044-50b2f17d-76d0-4fa3-9a41-d2d3087302ff.png)

It's internal api, so some document should help for opensource contribution.

**Comment by johanandren** [comment]
Makes sense, it's there as an extension point for Lightbend Telemetry to follow elements through a stream when they are suspended/put in buffers and then are introduced back into the stream.
